web/react/utils/utils.jsx

```diff
@@ -420,12 +491,32 @@ module.exports.textToJsx = function(text, options) {
         var highlightSearchClass = "";
         for (var z = 0; z < words.length; z++) {
             var word = words[z];
-            var trimWord = word.replace(puncStartRegex, '').replace(puncEndRegex, '').trim();
+            var trimWord;
+            if (useMarkdown) {
+                trimWord = word.replace(endTagRegex, '').replace(startTagRegex, '').replace(puncStartRegex, '').replace(puncEndRegex, '').trim();
+            } else {
+                trimWord = word.replace(puncStartRegex, '').replace(puncEndRegex, '').trim();
+            }
             var mentionRegex = /^(?:@)([a-z0-9_]+)$/gi; // looks loop invariant but a weird JS bug needs it to be redefined here
             var explicitMention = mentionRegex.exec(trimWord);
+            console.log(word);
+            var prefix;
+            var suffix;
+            var prefixSpan;
+            var suffixSpan;
+            if (useMarkdown) {
+                prefix = (word.match(startTagRegex) ? word.match(startTagRegex) : "") + (word.replace(startTagRegex, '').match(puncStartRegex) ? word.replace(startTagRegex, '').match(puncStartRegex) : "");
+                suffix = (word.match(endTagRegex) ? word.match(endTagRegex) : "") + (word.replace(endTagRegex, '').match(puncEndRegex) ? word.replace(endTagRegex, '').match(puncEndRegex) : "");
+                prefixSpan = prefix ? (<span key={word+i+z+"pre_span"}><span dangerouslySetInnerHTML={{__html: prefix}} /></span>) : null;
+                suffixSpan = suffix ? (<span key={word+i+z+"suf_span"}><span dangerouslySetInnerHTML={{__html: suffix}} /> </span>) : (<span key={word+i+z+"suf_span"}> </span>);
```

Comment:
I'm having difficulty understanding this part (which may mean that it should be changed), but if I'm understanding it correctly, if marked outputs something like "<b>...</b>", this will turn it into something like "<span><span><b></span></span>...<span><span></b></span></span>" which doesn't seem like it should work.